### PR TITLE
feat: automatically scale down Deployment after migrating to Rollout

### PR DESCRIPTION
**Pull request by balasoiuroxana**
fixes: https://github.com/argoproj/argo-rollouts/issues/2748

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-rollouts/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/) with a list of types and scopes found [here](https://github.com/argoproj/argo-rollouts/blob/master/.github/workflows/pr-title-check.yml), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed my commits with [DCO](https://github.com/argoproj/argoproj)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My builds are green. Try syncing with master if they are not. 
* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-rollouts/blob/master/USERS.md).

### CODE REVIEW

**Comment by dharma-rise** [comment]
Any updates?

**Comment by zachaller** [comment]
Can we add some documentation [here](https://argo-rollouts.readthedocs.io/en/stable/features/specification/) and [here](https://argo-rollouts.readthedocs.io/en/stable/migrating/#reference-deployment-from-rollout) on the behavior of all the scaledown options etc?

File: rollout/sync.go [hunk 1]
```diff
@@ -378,6 +378,12 @@ func (c *rolloutContext) scaleReplicaSet(rs *appsv1.ReplicaSet, newScale int32,
 			revision, _ := replicasetutil.Revision(rs)
 			c.recorder.Eventf(rollout, record.EventOptions{EventReason: conditions.ScalingReplicaSetReason}, conditions.ScalingReplicaSetMessage, scalingOperation, rs.Name, revision, oldScale, newScale)
 			c.replicaSetInformer.GetIndexer().Update(rs)
+			if rollout.Spec.WorkloadRef != nil && rollout.Spec.WorkloadRef.ScaleDown == v1alpha1.ScaleDownProgressively {
```
Comment (balasoiuroxana) [inline rollout/sync.go:381]: I realized that for the `ScaleDownProgressively` option the deployment will be scaled down without waiting  for the rollout replicas to become available.
Do you have any suggestions where we can test that? We should scale down the deployment only after the replicas become available.
cc: @carlossg

Comment (balasoiuroxana) [reply rollout/sync.go]: I changed the logic for the `progressively` scaleDown option, now the deployment is scaled down as the rollout is scaled up and its replicas become ready.

**Comment by zachaller** [comment]
Is this ready for a second review, now?

**Comment by balasoiuroxana** [comment]
@zachaller , yes it is ready

**Review by zachaller — APPROVED** [review]

**Comment by zachaller** [comment]
@balasoiuroxana looks good and ready to merge, can you fix the last small conflict?

**Comment by Ahmed-Elkollaly** [comment]
@balasoiuroxana @zachaller Thanks for this awesome feature but I don't see it in the latest argo rollout version v1.6.4. when it's expected to be rolled out?

**Comment by zachaller** [comment]
It will be available in 1.7 we don't have a set date on that yet, I want to try and get step plugins in 1.7

**Comment by rmn-lux** [comment]
> It will be available in 1.7 we don't have a set date on that yet, I want to try and get step plugins in 1.7

Hello, is there any information about the release date of v1.7?

**Comment by SleepyBrett** [comment]
Couldn't you also do this the other way as well.

On deletion of the rollout the reconciler takes the current replicas of the rollout and applies it to the referenced deployment (if available). Once the pods become healthy, delete the rollout's child replica-sets and remove the finalizer?

**Pull request merged**